src/main/java/bio/terra/service/filedata/flight/ingest/IngestDriverStep.java

```diff
@@ -0,0 +1,279 @@
+package bio.terra.service.filedata.flight.ingest;
+
+import bio.terra.model.BulkLoadArrayResultModel;
+import bio.terra.model.BulkLoadFileResultModel;
+import bio.terra.model.BulkLoadResultModel;
+import bio.terra.model.FileLoadModel;
+import bio.terra.service.filedata.exception.FileSystemCorruptException;
+import bio.terra.service.filedata.exception.FileSystemExecutionException;
+import bio.terra.service.filedata.flight.FileMapKeys;
+import bio.terra.service.job.JobMapKeys;
+import bio.terra.service.load.LoadCandidates;
+import bio.terra.service.load.LoadFile;
+import bio.terra.service.load.LoadService;
+import bio.terra.service.load.flight.LoadMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.FlightState;
+import bio.terra.stairway.Stairway;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.DatabaseOperationException;
+import bio.terra.stairway.exception.FlightNotFoundException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+// The driver step is the core part of the bulk load meta-flight. It runs a loop that keeps some
+// K flights busy performing file loads.
+//
+// We get these elements from the inputs
+// - DATASET_ID dataset we are loading into
+// - LOAD_TAG is the load tag for this ingest
+// - REQUEST is a BulkLoadArrayRequestModel
+//
+// It expects the following working map data:
+// - LOAD_ID - load id we are working on
+//
```

Comment:
I like the pattern of including the input and working map keys that the flight operates on in the comments here.